test/stdlib/NSStringAPI.swift

```diff
@@ -68,6 +68,50 @@ func createNSStringTemporaryFile()
   return (existingPath, nonExistentPath)
 }
 
+func createMutableStringVariants(_ str: NSString) -> [NSMutableString] {
+  let mutable = str.mutableCopy() as! NSMutableString
+  let mutableSlow = str.mutableCopy() as! NSMutableString
+  mutableSlow.append("𐐷")
+  mutableSlow.deleteCharacters(in: NSMakeRange(mutableSlow.length - 2, 1))
+  mutableSlow.deleteCharacters(in: NSMakeRange(mutableSlow.length - 1, 1))
+  let cowImmutable = mutable.copy() as! NSString
+  let slowCoWImmutable = mutableSlow.copy() as! NSString
+  let utf8ptr = str.utf8String!
+  let plain = NSString(utf8String: utf8ptr)!
+  let bridged = plain as String
+  let rebridged = bridged as NSString
+  return [
+    mutable,
+    mutableSlow,
+    cowImmutable.mutableCopy() as! NSMutableString,
+    slowCoWImmutable.mutableCopy() as! NSMutableString,
+    plain.mutableCopy() as! NSMutableString,
+    rebridged.mutableCopy() as! NSMutableString
+  ]
+}
+
+func createStringVariants(_ str: NSString) -> [NSString] {
+  let mutable = str.mutableCopy() as! NSMutableString
+  let mutableSlow = str.mutableCopy() as! NSMutableString
+  mutableSlow.append("𐐷")
+  mutableSlow.deleteCharacters(in: NSMakeRange(mutableSlow.length - 2, 1))
+  mutableSlow.deleteCharacters(in: NSMakeRange(mutableSlow.length - 1, 1))
+  let cowImmutable = mutable.copy() as! NSString
+  let slowCoWImmutable = mutableSlow.copy() as! NSString
+  let utf8ptr = str.utf8String!
+  let plain = NSString(utf8String: utf8ptr)!
+  let bridged = plain as String
+  let rebridged = bridged as NSString
+  return [
+    mutable as NSString,
+    mutableSlow as NSString,
+    cowImmutable,
+    slowCoWImmutable,
+    plain,
+    rebridged
+  ]
+}
+
```

Comment:
Do you test the switching behavior for sub-scalar indices, or for invalid appends directly?

Reply:
yup!